qt/main.cpp

```diff
@@ -207,7 +198,21 @@ int main(int argc, char * argv[])
         screenshotParams->m_dpiScale = FLAGS_dpi_scale;
     }
 
-    qt::MainWindow::SetDefaultSurfaceFormat(apiOpenGLES3);
+
+    QSurfaceFormat fmt;
+    fmt.setAlphaBufferSize(8);
+    fmt.setBlueBufferSize(8);
+    fmt.setGreenBufferSize(8);
+    fmt.setRedBufferSize(8);
+    fmt.setStencilBufferSize(0);
+    fmt.setSamples(0);
+    fmt.setSwapBehavior(QSurfaceFormat::DoubleBuffer);
+    fmt.setSwapInterval(1);
+    fmt.setDepthBufferSize(16);
+#ifdef ENABLE_OPENGL_DIAGNOSTICS
+    fmt.setOption(QSurfaceFormat::DebugContext);
+#endif
+    QSurfaceFormat::setDefaultFormat(fmt);
```

Comment:
Do we need it here since you moved initialization code into MapWidget::initializeGL() ?
Or there is no problem to call setDefaultFormat twice?

Reply:
There is absolutely no limit how many times you can call `setDefaultFormat`, every time you call it it will be in effect for the next surface that is either created manually or automatically by Qt.